apps/trading/client-pages/swap/swap-container.tsx

```diff
@@ -0,0 +1,307 @@
+import {
+  useAssetsMapProvider,
+  type AssetFieldsFragment,
+} from '@vegaprotocol/assets';
+import {
+  getBaseAsset,
+  getQuoteAsset,
+  isSpot,
+  marketDataProvider,
+  useMarketsMapProvider,
+  type MarketDataFieldsFragment,
+  type MarketFieldsFragment,
+} from '@vegaprotocol/markets';
+import { VegaIcon, VegaIconNames, Link } from '@vegaprotocol/ui-toolkit';
+import { useVegaWallet } from '@vegaprotocol/wallet-react';
+import { useAccounts, type Account } from '@vegaprotocol/accounts';
+import { useT } from '../../lib/use-t';
+import {
+  addDecimal,
+  addDecimalsFormatNumber,
+  formatNumber,
+  removeDecimal,
+} from '@vegaprotocol/utils';
+import { OrderTimeInForce, OrderType, Side } from '@vegaprotocol/types';
+import { useVegaTransactionStore } from '@vegaprotocol/web3';
+import noop from 'lodash/noop';
+import { getNotionalSize } from '@vegaprotocol/deal-ticket';
+import { useDataProvider } from '@vegaprotocol/data-provider';
+import { AssetInput, SwapButton, PriceImpactInput } from './swap-form';
+import BigNumber from 'bignumber.js';
+
+const POLLING_TIME = 5000;
+
+const assetBalance = (
+  asset?: AssetFieldsFragment,
+  accounts?: Account[] | null
+) => {
+  if (!asset || !accounts) return undefined;
+  const account = accounts.find((a) => a.asset.id === asset.id);
+  return account?.balance;
+};
+
+export interface SwapFields {
+  baseId: string;
+  quoteId: string;
+  baseAmount: string;
+  quoteAmount: string;
+  priceImpactTolerance: string;
+}
+
+const derivePrice = (
+  marketData: MarketDataFieldsFragment | null,
+  side: Side,
+  toleranceFactor: number,
+  market: MarketFieldsFragment
+) => {
+  if (!marketData) return;
+  const price =
+    side === Side.SIDE_BUY
+      ? new BigNumber(marketData.bestOfferPrice).times(1 + toleranceFactor)
+      : new BigNumber(marketData.bestBidPrice).times(1 - toleranceFactor);
+
+  return price;
+};
+
+export const SwapContainer = () => {
+  const t = useT();
+  const { watch, setValue, handleSubmit } = useForm<SwapFields>();
+  const { baseAmount, quoteAmount, priceImpactTolerance } = watch();
+
+  const { pubKey, isReadOnly } = useVegaWallet();
+  const { data: markets } = useMarketsMapProvider();
+  const { data: assetsData } = useAssetsMapProvider();
+
+  const create = useVegaTransactionStore((state) => state.create);
+
+  const [quoteAsset, setQuoteAsset] = useState<AssetFieldsFragment>();
+  const [baseAsset, setBaseAsset] = useState<AssetFieldsFragment>();
+  const [side, setSide] = useState<Side>();
+  const [marketId, setMarketId] = useState<string>('');
+  const [market, setMarket] = useState<MarketFieldsFragment>();
+
+  const { data: marketData, reload: reloadMarketData } = useDataProvider({
+    dataProvider: marketDataProvider,
+    variables: { marketId },
+    skip: !marketId,
+  });
+  const { data: accounts, reload: reloadAccounts } = useAccounts(pubKey);
+
+  useEffect(() => {
+    const intervalAccounts = setInterval(reloadAccounts, POLLING_TIME);
```

Comment:
We shouldn't need to poll it accounts, the data provider has a subscription. If its not always got the latest data something is broken elsewhere

Reply:
updated. removed polling altogether